### DISCUSSION TITLE
When you try to deploy Perses on k0s the container crashes with "too many open files"

### DISCUSSION THREAD

**Discussion by indigo423** [perses/perses#2159] [Q&A] [answered]
## Problem
I tried to use the Helm chart with the latest version just bumping the version number to v0.47.0.

The container crashes with the following error message:

```
time="2024-08-08T10:49:39Z" level=warning msg="encryption_key is not provided and therefore it will use a default one. For production instance you should provide the key."
time="2024-08-08T10:49:39Z" level=fatal msg="unable to initialize the service manager: open /etc/perses/schemas/panels: no such file or directory"
failed to create fsnotify watcher: too many open files%
```

## Investigation

I've switched to the v0.47.0-debug image and ran a debug like

```
kubectl debug -it -n app-perses pods/perses-0 --copy-to=perses-debug --container=perses -- sh
```

I've investigated the `max_user_watches` and `max_user_instances` which gave me

```
/etc/perses $ cat /proc/sys/fs/inotify/max_user_watches
61034

/etc/perses $ cat /proc/sys/fs/inotify/max_user_instances
128
```

The ulimit for open files is 

```
/perses $ ulimit -n
999999
```

## Solution

I was able to solve the problem by increasing the 

```
sysctl fs.inotify.max_user_instances = 1280
sysctl fs.inotify.max_user_watches = 655360
```

on my Kubernetes hosts to solve this problem.

**Comment by indigo423** [comment, marked as answer]
With a hint from @Nexucis, the reason might have been the wrong path for schema files caused watching for more files than necessary. I saw this problem while I was working on this PR https://github.com/perses/helm-charts/pull/16.

With the correct path to the schema folders, that workaround isn't necessary anymore and I was able to set it back to the defaults.

**Comment by Nexucis** [comment]
Thank you @indigo423 for the details about your issues and how you fixed it. Closing now the discussion as it's resolved
